public/hooks/use_paragraphs.ts

```diff
@@ -229,19 +234,43 @@ export const useParagraphs = () => {
           return;
         }
 
-        const para = context.state.getParagraphsValue()[index];
+        const paragraphs = context.state.getParagraphsValue();
+        const para = paragraphs[index];
         const isSavedObjectNotebook = isValidUUID(openedNoteId);
         context.state.value.paragraphs[index].updateUIState({
           isRunning: true,
         });
 
+        let contextPrompt: string = '';
+        if (
+          isSavedObjectNotebook &&
+          (para.input.inputType === DEEP_RESEARCH_PARAGRAPH_TYPE ||
+            para.input.inputType === AI_RESPONSE_TYPE)
+        ) {
+          try {
+            contextPrompt = await generateContextPromptFromParagraphs({
+              paragraphService,
+              paragraphs,
+              notebookInfo: topContextState.value,
+              ignoreInputTypes:
+                para.input.inputType === AI_RESPONSE_TYPE ? [] : [DEEP_RESEARCH_PARAGRAPH_TYPE],
+            });
+          } catch (err) {
+            notifications.toasts.addDanger(`Error running paragraph: ${err.message}`);
+            context.state.value.paragraphs[index].updateUIState({
+              isRunning: false,
+            });
+            return Promise.reject('Generate context failed');
+          }
+        }
+
         const paraUpdateObject = {
           noteId: openedNoteId,
           paragraphId: para.id,
           input: {
             inputType: para.input.inputType,
             inputText: para.input.inputText,
-            parameters: para.input.parameters,
+            parameters: { ...(para.input.parameters || {}), contextPrompt },
```

Comment:
```suggestion
            parameters: { ...(para.input.parameters || {}), PERAgentContext: contextPrompt },
```
The parameters is also consumed in PER agent paragraph, please check that.

Reply:
Thanks you point it, updated~